webhook/core/annotation_validation_webhook.go

```diff
@@ -84,6 +84,31 @@ func (av *AnnotationValidator) Handle(ctx context.Context, req admission.Request
 			}
 		}
 	}
+
+	// check if pod-sg annotation is updated by any resource
+	if podSgAnnot, ok := pod.Annotations["fargate.amazonaws.com/pod-sg"]; ok {
+		webhookLog.V(1).Info("Got annotation:", "fargate.amazonaws.com/pod-sg", podSgAnnot)
+		if req.Operation == v1beta1.Update {
+			webhookLog.V(1).Info("Operation is update, checking that pod-sg annotation wasn't modified")
+			oldPod, err := av.K8sWrapper.GetPodFromAPIServer(pod.Namespace, pod.Name)
+			if err != nil {
+				webhookLog.Error(err, "Failed to fetch pod in update request")
+				return admission.Denied("Validation failed. Trying to update annotation on a pod that can't be found")
+			}
+			if oldPodAnnotationValue, ok := oldPod.Annotations["fargate.amazonaws.com/pod-sg"]; ok {
+				// deny the request if pod-sg annotation is updated
+				if oldPodAnnotationValue != podSgAnnot {
```

Comment:
If you do decide to switch to get the pod object form cache instead. This is the sequence of event that may lead to an update to fail when it should go through.

Consider the sequence of events for a pod with same namespace/name. 

- Pod is terminating.
- Pod event received with deletion timestamp set.
- New Pod is created with same namespace/name.
- The cache could return the older pod (that is already deleted) which may have a different annotation then the current pod and then the event would fail.

Even though this is a rare scenario and not sure you could hit this in webhooks. But it would be worth adding extra guards like checking the UID when comparing older pod to a newer pod.

Reference: This kind of bug bit us in another controller and it's super hard to reproduce.